src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerOptions.cs

```diff
@@ -8,16 +8,33 @@ namespace Microsoft.AspNetCore.Server.Kestrel
 {
     public class KestrelServerOptions
     {
+        /// <summary>
+        /// Gets or sets whether the <c>Server</c> header should be included in each response.
+        /// </summary>
+        public bool AddServerHeader { get; set; } = true;
+
         public IServiceProvider ApplicationServices { get; set; }
 
         public IConnectionFilter ConnectionFilter { get; set; }
 
-        public bool NoDelay { get; set; } = true;
+        private int _maxInputBufferLength = 1024 * 1024;
+        public int MaxInputBufferLength
```

Comment:
Not that we are consistent with this, but we should add a doc comment. Probably something along the lines of `Per-request buffer limit measured in bytes.`